RationaleMCP/Changes/3_7/3610.md

```diff
@@ -0,0 +1,28 @@
+# Discrete-time variability based on `GenerateEvents` #3610
+
+## Main change
+
+Variability of functions with `GenerateEvents`.
+
+## Reason
+
+Making it possible to create functions that generate events similarly as the built-in functions, and have rules consistent with them just being inlined.
+
+## Breaks
+
+``` modelica
+function foo
+  input Real x;
+  output Integer i=bar(x)+(if x>0 then 1 else 0);
+  annotation(GenerateEvents=true);
+end foo;
+
+function bar
+  input Real x;
+  output Integer i=integer(x);
+  annotation(GenerateEvents=false);
+end bar;
```

Comment:
With the call to `foo` being last in the listing, I think the listing should consistently define functions before they are used:

Also, I am not sure whether the `(if x>0 then 1 else 0)` really helps pinpointing the problem…

```suggestion
function bar
  input Real x;
  output Integer i = integer(x);
  annotation(GenerateEvents=false);
end bar;

function foo
  input Real x;
  output Integer i = bar(x);
  annotation(GenerateEvents=true);
end foo;
```

Reply:
True, the `(if x>0 then 1 else 0)` was intended to make the example more realistic in some sense. (As the GenerateEvents=true was intended to generate events for that part, but realism might not be required here.)